scylla/src/transport/iterator.rs

```diff
@@ -535,46 +537,97 @@ where
     }
 }
 
-/// Iterator over rows returned by paged queries\
-/// Allows to easily access rows without worrying about handling multiple pages
-pub struct LegacyRowIterator {
-    current_row_idx: usize,
-    current_page: Rows,
+/// An intermediate object that allows to construct an iterator over a query
+/// that is asynchronously paged in the background.
+///
+/// TODO: implement and describe the new API
+///
+/// A pre-0.15.0 interface is also available, although deprecated:
+/// `into_legacy()` method converts RawLendingStream to LegacyRowIterator,
+/// enabling Stream'ed operation on rows being eagerly deserialized
+/// to a middle-man Row type. This is inefficient, especially if
+/// the Row type is not the intended target type.
+pub struct RawLendingStream {
```

Comment:
Is the TODO going to be done in this PR or follow up?

Reply:
It's addressed later in this PR

Reply:
To be exact - commit `iterator: typed API for new deserialization framework`